### DISCUSSION TITLE
Any examples of how to filter document listings by predefined properties?

### DISCUSSION THREAD

**Discussion by mrpdean** [pimcore/pimcore#10623] [Q&A / Support] [answered]
I'm wanting to create a basic blog on my Pimcore site.

I'd like to use documents (pages) for the blog articles so that the content authors get all the flexibility of the content blocks (bricks) I've set up, however I'd like to link each article (page) and an author and on the front end. I'd also like to be able to filter articles by the different authors.

I was thinking of using a predefined property to associate the article (page) with it's author (data object).

However, I can't work out how to create a document listing filtered by the authors predefined property, other than just getting all documents and looping over them to check the author property.  I feel like there must be a more elegant solution to this but I haven't been able to find out yet.

What would be really great is if we could add custom fields to to pages, or even create custom document types with the additional fields we need.

If anyone has any suggestions I'd be most grateful.

Thanks

**Comment by ckemptner** [comment, marked as answer]
I'd recommend using a data object for configuration and a document for the layout 🙂 Similar like in our demo, but you also assign a document via a relationship to the object. So for each article you would then have one data object and one document.

**Reply by mrpdean** [reply]
Thanks. That sounds like it would work.  I'll give that a try :)
